tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,330 @@
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@pytest.mark.parametrize(
+    argnames=['interface', 'resource_to_delete'],
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'mgr'],
+            marks=pytest.mark.polarion_id("OCS-810")
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'mon'],
+            marks=pytest.mark.polarion_id("OCS-811")
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'osd'],
+            marks=pytest.mark.polarion_id("OCS-812")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mgr'],
+            marks=pytest.mark.polarion_id("OCS-813")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mon'],
+            marks=pytest.mark.polarion_id("OCS-814")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'osd'],
+            marks=pytest.mark.polarion_id("OCS-815")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mds'],
+            marks=pytest.mark.polarion_id("OCS-816")
+        )
+    ]
+)
+class TestResourceDeletionDuringMultipleDeleteOperations(ManageTest):
+    """
+    Delete ceph/rook pod while deletion of PVCs, pods and IO are progressing
+    """
+    num_of_pvcs = 25
+    pvc_size = 3
+
+    @pytest.fixture()
+    def setup_base(
+        self, interface, multi_pvc_factory, pod_factory
+    ):
+        """
+        Create PVCs and pods
+        """
+        pvc_objs = multi_pvc_factory(
+            interface=interface,
+            project=None,
+            storageclass=None,
+            size=self.pvc_size,
+            access_mode=constants.ACCESS_MODE_RWO,
+            status=constants.STATUS_BOUND,
+            num_of_pvc=self.num_of_pvcs,
+            wait_each=False
+        )
+
+        pod_objs = []
+        for pvc_obj in pvc_objs:
+            pod_obj = pod_factory(pvc=pvc_obj, status="")
+            pod_objs.append(pod_obj)
+        for pod_obj in pod_objs:
+            wait_for_resource_state(
+                resource=pod_obj, state=constants.STATUS_RUNNING
+            )
+
+        return pvc_objs, pod_objs
+
+    def delete_pods(self, pods_to_delete):
+        """
+        Delete pods
+        """
+        for pod_obj in pods_to_delete:
+            pod_obj.delete(wait=False)
+        return True
+
+    def test_disruptive_during_pod_pvc_deletion_and_io(
+        self, interface, resource_to_delete,
+        setup_base
+    ):
+        """
+        Delete ceph/rook pod while PVCs deletion, pods deletion and IO are
+        progressing
+        """
+        pvc_objs, self.pod_objs = setup_base
+        sc_obj = pvc_objs[0].storageclass
+        namespace = pvc_objs[0].project.namespace
+
+        num_of_pods_to_delete = 10
+        num_of_io_pods = 5
+        pods_to_delete = self.pod_objs[:num_of_pods_to_delete]
+        io_pods = self.pod_objs[num_of_pods_to_delete:num_of_pods_to_delete + num_of_io_pods]
```

Comment:
please add comment clarifying what you are doing in this line

Reply:
Done